src/rs.cpp

```diff
@@ -2965,8 +2965,21 @@ void rs2_update_firmware_cpp(const rs2_device* device, const void* fw_image, int
 
     VALIDATE_NOT_NULL(device);
     VALIDATE_NOT_NULL(fw_image);
+
+    // TODO: DFU issue - remove d500_device usage when HKR will fix image size issue 
+    // or by asking if we are working with D500 device (including D5XX recovery device when it will be added).
+    // Meanwhile, d500_device is always true, but in the future we need to check some PIDs (see commented lines for example)
+    // if it equals to D585, D585S or D5XX_RECOVERY device, then pass the image size check
+    bool d500_device = true;
+    /*if (device->device->supports_info(RS2_CAMERA_INFO_PRODUCT_ID))
```

Comment:
The comment lines looks better, can we use them?

Reply:
We get here as d400 recovery device (D4XX)
No solution to this till HKR will support fixed image size or will identify itself as D5XX device